library/packages/src/modules/PackageCallbacks.rb

```diff
@@ -2487,6 +2487,110 @@ def ProcessDone
+      elsif UI.WidgetExists(:progressCurrentPackage)
+        UI.ChangeWidget(Id(:progressCurrentPackage), :Value, progress)
+      else
+        UI.ChangeWidget(Id(:pb), :Value, progress)
+      end
+
+      ui = UI.PollInput
+      ui != :abort && ui != :cancel
+    end
+
+    # Handle the file conflict detection result callback.
+    # Ask to user whether to continue. In the AutoYaST mode an error is reported
+    # but the installation will continue ignoring the confliucts.
+    # @param excluded_packages [Array<String>] packages ignored in the check
+    #   (e.g. not available for check in the download-as-needed mode)
+    # @param conflicts [Array<String>] list of translated descriptions of
+    #   the detected file conflicts
+    # @return [Boolean] true = continue, false = abort
+    def FileConflictReport(excluded_packages, conflicts)
+      log.info "Excluded #{excluded_packages.size} packages in file conflict check"
+      log.debug "Excluded packages: #{excluded_packages.inspect}"
+      log.info "Found #{conflicts.size} conflicts: #{conflicts.join("\n\n")}"
+
+      # just continue installing packages if there is no conflict
+      return true if conflicts.empty?
+
+      # don't ask in autoyast or command line mode, just report/log the issues and continue
+      if Mode.autoinst || Mode.autoupgrade || Mode.commandline
+        # TRANSLATORS: An error message, %s is the actual list of detected conflicts
+        Report.Error(_("File conflicts detected, the conflicting files will " \
+          "be overwritten:\n\n%s") % conflicts.join("\n"))
+        return true
+      end
+
+      button_box = ButtonBox(
+        PushButton(Id(:continue), Opt(:default, :okButton), Label.ContinueButton),
+        PushButton(Id(:cancel), Opt(:cancelButton), Label.CancelButton)
+      )
+
+      # TRANSLATORS: A popup label, use max. 70 chars per line, use more lines if needed
+      label = _("File conflicts happen when two packages attempt to install\n" \
+        "files with the same name but different contents. If you continue,\n" \
+        "the conflicting files will be replaced losing the previous content.")
+
+      # TRANSLATORS: Popup heading
+      heading = n_("A File Conflict Detected", "File Conflicts Detected", conflicts.size)
+
+      dialog = VBox(
+        Left(Heading(heading)),
+        VSpacing(0.2),
+        Left(Label(label)),
+        MinSize(65, 15, RichText(Opt(:plainText), conflicts.join("\n\n"))),
+        button_box
+      )
+
+      UI.OpenDialog(dialog)
+
+      begin
+        UI.SetFocus(Id(:continue))
+        ret = UI.UserInput
+        log.info "User Input: #{ret}"
+        ret == :continue
+      ensure
+        UI.CloseDialog
+      end
+    end
+
+    # Handle the file conflict detection finish callback.
+    def FileConflictFinish
+      log.info "File conflict check finished"
+      return if Mode.commandline
+
+      Progress.Finish unless UI.WidgetExists(:progressCurrentPackage)
+    end
```

Comment:
Btw I think this class is already too big. So at least for this case I suggest to use its namespace like Yast::FileCallbacks and inside it define methods. Ideally in own file. And just use require and Set method here.